src/libcore/hash/mod.rs

```diff
@@ -572,6 +570,9 @@ mod impls {
                 fn hash_slice<H: Hasher>(data: &[$ty], state: &mut H) {
                     let newlen = data.len() * mem::size_of::<$ty>();
                     let ptr = data.as_ptr() as *const u8;
+                    // SAFETY: `ptr` is valid and aligned, the new slice only spans
+                    // across `data` and is never mutated, and its total size is the
+                    // same as the original `data` so it can't be over `isize::MAX`.
                     state.write(unsafe { slice::from_raw_parts(ptr, newlen) })
```

Comment:
This is technically insufficient. There may be padding bytes -- which are uninitialized, and so `&[u8]` isn't permitted to contain them. However, that is not be a problem here as impl_write! expands only with numeric primitives which don't have padding. The safety comment should say so, though.